### Exercises

**Our test function:**

```
function foo (shouldFail){
  return function (errCb, successCb) {
    shouldFail ? errCb(new Error('What were the chances?')) : true
  }
}
```

**Some helpers**
```
function success (successful) {
  console.log("It was a success! We got", successful)
}

function failure (error) {
  console.log("It all went wrong! We got", error)
}
```

```
function addLog (eitherErrorOrSuccess) {

}
```

```
function capitaliseErrorMessage (eitherErrorOrSuccess) {

}
```

**Test**: `addLog(capitaliseErrorMessage(foo(true)))(failure, success)`

```
function logSuccessfulValue (eitherErrorOrSuccess) {

}
```
**Test**: logSuccessfulValue(foo(fail))(failure, success)

```
function try2PossibleMistakes (eitherErrorOrSuccess, somethingThatMightFail) {

}
```

**Test**:
```
try2PossibleMistakes(foo(true), foo)(failure, success)
try2PossibleMistakes(foo(false), foo)(failure, success)
```
